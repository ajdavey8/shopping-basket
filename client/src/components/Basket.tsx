import React from 'react';
import { Link } from 'react-router-dom';

interface Basket {
  total: number
  numberOfItems: number
}

const Basket = ({ total, numberOfItems }: Basket) => (
  <section>
    <h4>Your Basket</h4>
    <h2>Â£{total}</h2>
    <p>{numberOfItems} item{numberOfItems !== 1 && 's'}</p>
    <Link to='/checkout'>Checkout </Link>
  </section>
)

export default Basket