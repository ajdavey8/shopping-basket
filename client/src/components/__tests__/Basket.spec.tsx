import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react'
import Basket from '../Basket';

const setup = () => {
  const utils = render(
    <Router>
      <Basket
        total={3}
        numberOfItems={2}
      />
    </Router>
  )

  return {
    ...utils
  }
}

describe('<Basket>', () => {
  it('renders with the correct props', () => {
    const { asFragment } = setup()
    expect(asFragment).toMatchSnapshot();
  })
})