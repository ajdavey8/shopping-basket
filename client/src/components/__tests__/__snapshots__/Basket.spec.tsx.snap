// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Basket> renders with the correct props 1`] = `[Function]`;
